### PR TITLE
V7.35: FIRMWARE_VERSION single source of truth + docs-drift audit (#124)

### LINKED ISSUE #124: Single source of truth for firmware version + docs-drift audit (README vs code)
## Problem

**Version string is scattered.** "V7.34" lives in the sketch header comment (rc_test.ino:2), the `debugInit()` banner (rc_test.ino:1462 — a 300-char changelog-in-a-string), CLAUDE.md (2 places), and implicitly in FIRMWARE-UPLOAD-LOG rows. Nothing ties them together, so they drift.

**Docs have drifted from code.** Found during review:

- README.md says joystick throttle boost is "×1.05" — code says `JOY_THROTTLE_GAIN = 1.40f` (rc_test.ino:210).
- README.md uses gear name "Turbo" — code and CLAUDE.md use "Boost".
- README pivot cap "60%" is right for Normal/Boost but Eco is 72.5% since #72 (`PIVOT_SPEED_CAP_LOW`).

## Proposal

1. `const char FIRMWARE_VERSION[] = "V7.34";` in `config.h` (post-split) — used by the debug banner, the SSE/JSON payload (dashboard can display it), and referenced by the ETag builder. The banner should be `version + short feature tag`, not a changelog (the changelog is git + FIRMWARE-UPLOAD-LOG).
2. One audit pass over README.md / OPERATOR-GUIDE.md / PROJECT-PLAN.md against current `[CONFIG]` values; fix drift.
3. Add a rule to CLAUDE.md: user-facing docs quote *behavior*, not tunable numbers, unless the doc is the tuning reference — numbers belong to `[CONFIG]` and the operator guide only.

## Acceptance

- [ ] Version string defined exactly once in code
- [ ] Dashboard shows the running firmware version (nice-to-have)
- [ ] README/operator-guide numbers match code or are removed in favor of behavior descriptions

Part of epic #116. Standalone, low-risk.

### PR DESCRIPTION
**Pull request by boulaajaj**
Closes #124

## Summary

The firmware version now has a **single source of truth**, and the README/PROJECT-PLAN drift the issue documented is fixed.

**Role tag:** `[C-Builder]`

**Code (V7.35):**
- `const char FIRMWARE_VERSION[] = "V7.35"` at the top of `[CONFIG]` — the only definition. Bump here, nowhere else.
- `debugInit()` banner is now `version + short tag` instead of a 300-char changelog-in-a-string (changelog belongs to git history + FIRMWARE-UPLOAD-LOG). **This observable banner change is explicitly authorized by #124's own proposal** — no other behavior is touched (banner is debug-serial only, not asserted by any test).
- Dashboard version display: **deferred** (issue marks it nice-to-have; it would touch the SSE frame budget and require the dashboard UI-test cycle — a clean standalone change if wanted later).

**Docs audit (against current `[CONFIG]`):**
- README: `Turbo`→`Boost` (×2); the **wrong** "×1.05" joystick-gain claim (actual `JOY_THROTTLE_GAIN` differs) and **stale pre-#113 reverse-cap numbers** (50%/62.5%) replaced with behavior descriptions + pointers to `[CONFIG]`/OPERATOR-GUIDE
- PROJECT-PLAN: stale "Status (V7.14)" header and file-map version replaced with SSOT pointers; stale historical ×1.05 line de-numbered
- OPERATOR-GUIDE: audited — its tuning table already matches code exactly (no changes)
- New CLAUDE.md rule: **README describes behavior, never tunable numbers**; PROJECT-PLAN/OPERATOR-GUIDE are technical references where numbers are allowed but must match `[CONFIG]`

## Test plan

- `arduino-cli` compile clean (44% flash)
- Host suite green **92/92** (no expectations touched; banner unasserted — commit via `DIGGER_NO_TEST_CHANGE=1` per the gate's design for test-neutral edits)
- All CI green
- Bench note: first flash after merge shows the new one-line banner — FIRMWARE-UPLOAD-LOG row will record V7.35

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated the startup/debug banner and documentation to reflect the current firmware version consistently.

* **Documentation**
  * Renamed **Turbo** to **Boost** throughout operator guidance.
  * Revised joystick/gear-cap explanations to remove outdated hardcoded percentages and numeric boost claims, replacing them with descriptions that reference current configuration values.
  * Added documentation guidance to prevent future version/tuning drift across files.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -188,7 +188,8 @@ lastGoodMs, valid }`. The `[TELEMETRY]` module polls it on Serial1 (D0/D1):
 
 ## Architecture Summary
 
-Sketch: `sketches/rc_test/rc_test.ino` (V7.34 — GL10 FOC + telemetry + Wi-Fi + alarms + battery/thermal safety) + `types.h` + `web_page.h`
+Sketch: `sketches/rc_test/rc_test.ino` (V7.35 — GL10 FOC + telemetry + Wi-Fi + alarms + battery/thermal safety) + `types.h` + `web_page.h`. The
+version string is defined ONCE: `FIRMWARE_VERSION` at the top of `[CONFIG]` (#124).
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply CLAUDE.md]: Accepted — fixed in 1b20296. Fair catch: this PR was reintroducing exactly the scatter it exists to kill. The architecture-summary line now describes the sketch without a version literal and states that FIRMWARE_VERSION in [CONFIG] is the only place the version exists (no doc carries a copy).

File: CLAUDE.md [hunk 2]
```diff
@@ -292,7 +293,7 @@ and the Arduino-side filter was double-smoothing the stream (operator felt
 ```text
 PROJECT-PLAN.md                          — Full technical specification
 OPERATOR-GUIDE.md                        — User guide for Jason (RC) and Malaki (joystick)
-sketches/rc_test/rc_test.ino             — Main Arduino sketch (V7.34 — GL10 FOC + telemetry + Wi-Fi + alarms + safety)
+sketches/rc_test/rc_test.ino             — Main Arduino sketch (V7.35 — GL10 FOC + telemetry + Wi-Fi + alarms + safety)
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply CLAUDE.md]: Accepted — fixed in 1b20296. The file-map row now reads 'version: FIRMWARE_VERSION in [CONFIG]' instead of embedding the literal. Historical V7.x mentions elsewhere in CLAUDE.md (implementation-status checkboxes, hardware history) are dated records and stay.

File: docs/DECISION-LOG.md [hunk 3]
```diff
@@ -5,6 +5,25 @@ Updated by session hooks — only technical content, no personal info.
...
+- **Docs drift fixed (audited against [CONFIG]):** README "Turbo"→Boost
+  (twice); "×1.05" joystick boost claim was wrong (JOY_THROTTLE_GAIN is a
+  different value — README now describes behavior without the number);
+  stale reverse-cap percentages (pre-#113 50%/62.5%) removed; gear table
+  de-numbered. PROJECT-PLAN stale "Status (V7.14)" header + file-map
+  version replaced with pointers to the SSOT.
+- **New rule (CLAUDE.md Style):** user-facing docs (README, PROJECT-PLAN)
+  describe behavior, never tunable numbers — numbers live only in [CONFIG]
+  and OPERATOR-GUIDE (the tuning reference). OPERATOR-GUIDE table audited:
+  already correct (65/80/100, 55/55/65, 72.5/60/60).
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply docs/DECISION-LOG.md]: Accepted — fixed in 1b20296. The DECISION-LOG bullet was written before the rule was refined mid-PR and contradicted the final CLAUDE.md wording. Since the entry is part of this unmerged PR, correcting it is normal revision (merged log entries would be immutable). Now consistent: README = behavior-only; PROJECT-PLAN and OPERATOR-GUIDE = technical references whose numbers must match [CONFIG].

**Comment by boulaajaj** [comment]
@coderabbitai review

File: sketches/rc_test/rc_test.ino [hunk 4]
```diff
@@ -77,6 +77,10 @@ UART sbusUart(PIN_SBUS_TX, PIN_SBUS_RX);
 // [CONFIG] — All tunable constants
 // ═══════════════════════════════════════════════════════════════
 
+// Firmware version — SINGLE SOURCE OF TRUTH (#124). The debug banner prints
+// it; docs and FIRMWARE-UPLOAD-LOG reference it. Bump here, nowhere else.
+const char FIRMWARE_VERSION[] = "V7.35";
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply sketches/rc_test/rc_test.ino:82]: Confirmed and fixed in 96de138 — excellent catch, this was a real gap between two things this epic shipped: the #129 checker matched only '#define FIRMWARE_VERSION', so the const char form this PR introduces was invisible to the SSOT gate. Detection is now a multiline regex covering both forms (macro and const char), the self-test fixture defines the version once in each form and must report the duplicate, and the real-repo run now shows the gate ACTIVE (the 'activates with #124' note is gone — it sees exactly one definition).

File: CLAUDE.md [hunk 5]
```diff
@@ -188,7 +188,9 @@ lastGoodMs, valid }`. The `[TELEMETRY]` module polls it on Serial1 (D0/D1):
 
 ## Architecture Summary
 
-Sketch: `sketches/rc_test/rc_test.ino` (V7.34 — GL10 FOC + telemetry + Wi-Fi + alarms + battery/thermal safety) + `types.h` + `web_page.h`
+Sketch: `sketches/rc_test/rc_test.ino` (GL10 FOC + telemetry + Wi-Fi + alarms + battery/thermal safety) + `types.h` + `web_page.h`. The version
+string is defined ONCE — `FIRMWARE_VERSION` at the top of `[CONFIG]` (#124);
+no doc carries a copy.
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply CLAUDE.md]: Accepted — fixed in 96de138. Wording now distinguishes the authoritative live copy (code only) from dated records: 'No doc carries a live copy; dated records (DECISION-LOG, FIRMWARE-UPLOAD-LOG) cite versions historically.' That matches how the logs actually work — entries are immutable snapshots, not live state.

File: PROJECT-PLAN.md [hunk 6]
```diff
@@ -151,7 +151,11 @@ python live_plot.py
 
 ---
 
-## Status (V7.14)
+## Status
+
+> Current firmware version: `FIRMWARE_VERSION` in `rc_test.ino` `[CONFIG]`
```
[the remark this answers is not in the document]
Comment (boulaajaj) [reply PROJECT-PLAN.md]: Accepted — fixed in 96de138. The PROJECT-PLAN status blockquote now gives the full repo path (sketches/rc_test/rc_test.ino) so the SSOT location is unambiguous outside the File Map context.

**Comment by boulaajaj** [comment]
@coderabbitai review
